### ISSUE TITLE
Frequent crashes with crop and rotate and perspective (Win 10, openCL activated)

### ISSUE THREAD

**Issue by europlatus** [darktable-org/darktable#13726]
**Describe the bug/issue**
I'm getting frequent crashes with the crop module and rotate and perspective. The application crashes to desktop with no backtrace generated. With `-d all` activated, I have a log produced. the last entry is:
"Magick: caught exception 0xC0000005 "Access violation"...

Log file of last session attached to this report.

**To Reproduce**
I can't reproduce it at will, but it happens frequently with these two modules in particular. I have tried disabling openCL but the program becomes unusably slow.

**Which commit introduced the error**
I didn't have this issue before 4.2.0. It remains a problem with 4.2.1

**Platform**
Windows 10

* darktable version : 4,2,1
* OS                : Win10
* Memory            : 32GB
* Graphics card     : AMD Radeon RX 6700 XT
* Graphics driver   : Driver Version 22.20.29.10-221130a-386220E-AMD-Software-Adrenalin-Edition
* OpenCL installed  : Yes (API version 2.0 / OpenCL® Driver Version 31.0.12029.10015)
* OpenCL activated  : Yes

**Additional context**
 - Can you reproduce with another darktable version(s)? **yes with version 4.2.0**
 - Can you reproduce with a RAW or Jpeg or both? **both**
 - Are the steps above reproducible with a fresh edit (i.e. after discarding history)? **yes, although not always reproducible**
 - Do you use lua scripts? No

[darktable-log crash.txt](https://github.com/darktable-org/darktable/files/10820749/darktable-log.crash.txt)

**Comment by gi-man** [comment]
I made the mistake of downloading and trying to read this on my cell earlier today. 12mb file with 133k rows.

The -d all is just too much information to read what's going on. I would first suggest going with -d perf and maybe -d opencl. 

My first suggestions: 1) Install latest drivers from AMD for the  [GPU card](https://www.amd.com/en/support/graphics/amd-radeon-6000-series/amd-radeon-6700-series/amd-radeon-rx-6700-xt). 2) go and turn off Tune OpenCL Performance in preferences. 3) after that, post the results of darktable-cltest

**Comment by europlatus** [comment]
Sorry for the large file. I'll do as you suggest and report back with findings asap. Thank you!

**Comment by europlatus** [comment]
Here are the results of cltest.exe after updating to the latest drivers and turning off the OpenCL tune option in preferences
[cltest.txt](https://github.com/darktable-org/darktable/files/10827669/cltest.txt)

**Comment by gi-man** [comment]
That looks good. Now try to reproduce with -d perf.

**Comment by europlatus** [comment]
Just crashed with -d perf on, but interestingly it was color balance rgb that caused it to crash this time. Log attached. When I restarted darktable, all the edits on that image over the last 5-10 minutes or so had been lost.

I'm currently running it with -d opencl at the moment, but so far no crashes.
[darktable -perf crash.txt](https://github.com/darktable-org/darktable/files/10828438/darktable.-perf.crash.txt)

**Comment by europlatus** [comment]
And here's another log with -d opencl turned on. This time it crashed applying a style.
[darktable -opencl crash.txt](https://github.com/darktable-org/darktable/files/10829113/darktable.-opencl.crash.txt)

**Comment by ptilopteri** [comment]
try with opencl disabled
note: -d opencl does not disable opencl
use    --disable-opencl

**Comment by gi-man** [comment]
The image changes are not saved until you switch to lighttable (and I think when you go to the next image). The changes are lost during a crash. It looks like it crashed while creating a preview which is using the CPU path.

```390.654523 [dev] took 0.000 secs (0.000 CPU) to load the image.
390.815338 [dev_pixelpipe] took 0.000 secs (0.000 CPU) [preview] processed `white balance' on CPU, blended on CPU
390.817664 [dev_pixelpipe] took 0.002 secs (0.000 CPU) [preview] processed `highlight reconstruction' on CPU, blended on CPU
390.836340 [dev_pixelpipe] took 0.019 secs (0.234 CPU) [preview] processed `demosaic' on CPU, blended on CPU
390.949701 [dev_pixelpipe] took 0.113 secs (0.656 CPU) [preview] processed `denoise (profiled)' on CPU, blended on CPU
390.994770 [dev_pixelpipe] took 0.045 secs (0.562 CPU) [preview] processed `lens correction' on CPU, blended on CPU
391.000594 [dev_pixelpipe] took 0.006 secs (0.000 CPU) [preview] processed `orientation' on CPU, blended on CPU
391.035336 [dev_pixelpipe] took 0.035 secs (0.375 CPU) [preview] processed `exposure' on CPU, blended on CPU
391.040107 [dev_pixelpipe] took 0.005 secs (0.188 CPU) [preview] processed `crop' on CPU, blended on CPU
391.049971 [dev_pixelpipe] took 0.010 secs (0.000 CPU) [preview] processed `input color profile' on CPU, blended on CPU
image colorspace transform Lab-->RGB took 0.003 secs (0.000 CPU) [channelmixerrgb ]
391.079299 [dev_pixelpipe] took 0.029 secs (0.219 CPU) [preview] processed `color calibration' on CPU, blended on CPU
image colorspace transform RGB-->Lab took 0.001 secs (0.000 CPU) [atrous ]
image colorspace transform RGB-->Lab took 0.020 secs (0.172 GPU) [atrous ]
image colorspace transform Lab-->RGB took 0.003 secs (0.000 CPU) [atrous ]
image colorspace transform Lab-->RGB took 0.001 secs (0.000 CPU) [atrous ]
image colorspace transform Lab-->RGB took 0.009 secs (0.188 GPU) [atrous ]
image colorspace transform Lab-->RGB took 0.002 secs (0.000 GPU) [atrous ]
391.180062 [dev_pixelpipe] took 0.101 secs (0.906 CPU) [preview] processed `contrast equalizer' on CPU, blended on CPU
image colorspace transform RGB-->Lab took 0.001 secs (0.000 CPU) [atrous 1]
image colorspace transform Lab-->RGB took 0.003 secs (0.000 CPU) [atrous 1]
image colorspace transform Lab-->RGB took 0.003 secs (0.000 CPU) [atrous 1]
Magick: caught exception 0xC0000005 "Access violation"...

**Comment by europlatus** [comment]
I have been using darktable with openCL disabled for a couple of days and so far no crashes, I am running it with `-d perf` to capture anything useful if and when it does crash. The main problem now of course is that the program is running painfully slow.

**Comment by europlatus** [comment]
I am still running darktable with openCL disabled and still no crashes. But now I have just encountered another problem with the crop module. The picture went black on one of my images.  I could get the correct image back by adjusting the size of the window (1st screenshot), but when I'm using my full screen space (resolution:  3440x1440 ultrawide), it goes black as shown in the 2nd screenshot below. I don't know if this is linked to the crash problems I've had with the crop module, but including here in case it gives a clue to something...

![image](https://user-images.githubusercontent.com/89879133/221476396-77d18a15-19e7-407c-863c-f679d30a87be.png)

![image](https://user-images.githubusercontent.com/89879133/221476456-de1233ac-ec59-49de-9b3d-84bd4b6ac3e4.png)

**Comment by gi-man** [comment]
I think to be able to troubleshoot more, you will need to turn opencl back on. I would then use -d opencl. Yes, this means that you can loose your edits, so go back and forth between light table and darkroom to save some of the work.

**Comment by europlatus** [comment]
Ok will do. Default openCL settings or use one of the tuning options?

**Comment by gi-man** [comment]
Default

In the back image, any clues from the log file?

**Comment by europlatus** [comment]
> In the back image, any clues from the log file?

Not that I can see. I didn't have any debugging turned on when it first happened, but I then turned it on and opened the image again. The image was still black at the time, but it then sorted itself out after I turned a few modules on and off.
Here is what I captured.
[darktable -perf black image.txt](https://github.com/darktable-org/darktable/files/10845426/darktable.-perf.black.image.txt)
